websites/serializers.py

```diff
@@ -138,11 +133,46 @@ def validate_email(self, email):
     def validate(self, attrs):
         """Make sure all required attributes are present for post/patch"""
         if not attrs.get("role"):
-            raise ValidationError("Role is required")
+            raise ValidationError({"role": "Role is required"})
         if not self.instance and not attrs.get("email"):
-            raise ValidationError("Email is required")
+            raise ValidationError({"email": "Email is required"})
         return attrs
 
+    def create(self, validated_data):
+        """Creating a contributor adds the user to the group corresponding to the specified role"""
+        website = self.website
+        role = validated_data.get("role")
+        user = User.objects.get(email=validated_data.get("email"))
+        group_name = permissions_group_name_for_role(role, website)
+        if user in get_users_with_perms(website) or user == website.owner:
+            raise ValidationError(
+                {"errors": ["User is already a collaborator for this site"]}
```

Comment:
I think we should namespace this error under "email". If we do that, the error message will show under the email input rather than under the form, which I think makes more sense 

![ss 2021-04-13 at 14 04 07 ](https://user-images.githubusercontent.com/14932219/114599505-3206cf80-9c61-11eb-91fe-e18e80d65df9.png)